### ISSUE TITLE
Nested tabs inside stacknavigator with swipe enabled crashes app

### ISSUE THREAD

**Issue by ovaris** [react-community/react-navigation#1677]
### Current Behavior
Navigating from route A to route B at stackNavigator where route B screen is tabsNavigator (so tabs is nested inside stack) causes app to crash on IOS  without no clear error message in logs. Bug is ONLY happening when either `swipeEnabled=true` OR `animationEnabled=true` in `TabNavigator` config AND it only happens when I first have visited inside Tabs and go back to top level stackRoute and when trying to go again to Tabs --> crash. I wish this could be easier to explain. Any tips are highly appreciated!

Here's my navigators:

```javascript

const AppTabNavigator = TabNavigator(
    {
        TabOne: { screen: TabOne },
        TabTwo: { screen: TabTwo },
        TabThree: { screen: TabThree },
    },
    {
        swipeEnabled: true, //setting this to false prevents app crashing
        animationEnabled: true, //and this too
        tabBarPosition: 'bottom',
        tabBarOptions: {
            showLabel: true,
            showIcon: true,
        }
    },
);

const AppStackNavigator = StackNavigator(
    {
        Tabs: { screen: Tabs }, //into this one
        Foo: { screen: Foo }, //app crashes when navigating from this screen ^^
    }
);

export default class Tabs extends Component {
    static propTypes = {
        dispatch: PropTypes.func.isRequired,
        tabNav: PropTypes.object.isRequired,
    };

    static navigationOptions = {
        header: null
    };

    render() {
        const { dispatch } = this.props;
        return (
            <View
                style={{ flex: 1 }}
            >
              <AppNavigator navigation={addNavigationHelpers({ dispatch, state: this.props.tabNav })} />
            </View>
        );
    }
}
@connect(
  state => ({
      stackNav: state.stackNav,
      user: state.auth.user
  }),
  dispatch => ({ dispatch }),
)
export default class AppWithNavigationState extends Component {
    static propTypes = {
        dispatch: PropTypes.func.isRequired,
        stackNav: PropTypes.object.isRequired,
    };

    render() {
        const { dispatch, stackNav } = this.props;
        return (
          <View
            style={{ flex: 1 }}
          >
            <AppStackNavigator navigation={addNavigationHelpers({ dispatch, state: stackNav })} />
          </View>
        );
    }
}
```

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation |  beta.11
| react-native     |  0.44
| node             | 
| npm or yarn      |

**Comment by eflath** [comment]
i am also experiencing this with a similar setup. The js error is:
```
navigation.dispatch is not a function. (In 'navigation.dispatch(_NavigationActions2.default.back({ key: backFromScene.route.key }))', 'navigation.dispatch' is undefined)
```

**Comment by jslz** [comment]
Anecdotally, I am seeing something similar using StackNavigator, go to a Component that renders a WebView. When doing the swipe from left to right gesture to dismiss it off the top of the stack, once the animation is done and the previous card is showing we get this error. Only on a real iOS device, not on the simulator, both running a Debug build.

```
[error][tid:com.facebook.react.JavaScript] navigation.dispatch is not a function. (In 'navigation.dispatch(_NavigationActions2.default.back({ key: backFromScene.route.key }))', 'navigation.dispatch' is undefined)
```

**Comment by xiaoyongchen** [comment]
@jslz 你应该是自定义的router吧，使用redux自定义就是有这个问题，没办法只有把手势设置为false，貌似手势是写在里面的，不知道能不能继承，或者是自定义。

**Comment by kelset** [comment]
Pinging OP @ovaris since this issue is quite old and related to an old version of the lib. If no answer I'll close in 7 days.

**Comment by kelset** [comment]
Hi there @ovaris ,

In an effort to cleanup this project and prioritize a bit, since this issue had no follow up since my last comment I'm going to close it.

If you are still having the issue with the latest version (especially if it's a bug report) please check the current open issues, and if you can't find a duplicate, open a new one that uses the new Issue Template to provide some more details to help us solve it.

**Comment by samanthacampisi** [comment]
I know this issue is closed but I'm still getting the same problem with the same scenario.
<img width="399" alt="screen shot 2018-02-12 at 10 31 06" src="https://user-images.githubusercontent.com/5727783/36090240-dfd5419e-0fdf-11e8-8d4e-4ca6dd466e17.png">


Did anybody find a solution for this? @ovaris @xiaoyongchen @jslz

**Comment by brentvatne** [comment]
can someone who would like to see this issue resolved please create a new issue with a reproducible example on https://snack.expo.io or in a github repository, with a minimal amount of superfluous code? (please don't post your entire app, just a small one to reproduce the problem). thanks!
